src/main/java/ev3dev/actuators/lego/motors/BaseRegulatedMotor.java

```diff
@@ -206,16 +206,31 @@ public void hold() {
      * Cancels any rotate() orders in progress
      */
     public void stop() {
-		this.setStringAttribute(COMMAND, STOP);
-
-        for (RegulatedMotorListener listener : listenerList) {
-            listener.rotationStopped(this, this.getTachoCount(), this.isStalled(), System.currentTimeMillis());
-        }
+        stop(true);
     }
 
     @Override
-    public void stop(boolean b) {
+    public void stop(boolean immediateReturn) {
+        doStop(HOLD, immediateReturn);
+    }
+
+
+    /**
+     * Backend for all stop moves. This sets the stop action type and then triggers the stop action.
+     * @param mode One of BRAKE, COAST and HOLD string constants.
+     * @param immediateReturn Whether the function should busy-wait until the motor stops reporting the 'running' state.
+     */
+    private void doStop(String mode, boolean immediateReturn) {
+        this.setStringAttribute(STOP_COMMAND, mode);
         this.setStringAttribute(COMMAND, STOP);
+
+        if (!immediateReturn) {
+            waitComplete();
+        }

```

Comment:
This is a new code to honor the `immediateReturn` parameter. 

This is how it is implemented on leJOS:
1. https://sourceforge.net/p/lejos/ev3/code/ci/master/tree/ev3classes/src/lejos/internal/ev3/EV3MotorPort.java#l427 newMove() -> waitComplete()
2. https://sourceforge.net/p/lejos/ev3/code/ci/master/tree/ev3classes/src/lejos/internal/ev3/EV3MotorPort.java#l379 waitComplete() body